### PR TITLE
Added Windows/MacOSX CI for py3.7 only

### LINKED ISSUE #1079: Replace conda install in github CI by actions/setup-python@v2
## 🚀 Feature

In order to accelerate env and dependencies installation we can opt to replace:
https://github.com/pytorch/ignite/blob/cc78b3bbd764da9ae7dee0df21676b2b8a50fba8/.github/workflows/unittests.yml#L19-L25
by 
https://github.com/pytorch/ignite/blob/cc78b3bbd764da9ae7dee0df21676b2b8a50fba8/.github/workflows/tpu-tests.yml#L14-L19

with correct python version

### LINKED ISSUE #1111: Setup Windows CI with github actions
## 🚀 Feature

To prevent bugs like https://github.com/pytorch/ignite/issues/1110 we need to setup windows CI. For example as a github action.

https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#github-hosted-runners

### PR DESCRIPTION
**Pull request by vfdev-5**
Fixes #1111  and 
Fixes #1079

Description:
- Added Windows/MacOSX CI for py3.7 only + MNIST example

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

File: .github/workflows/unittests.yml [hunk 1]
```diff
@@ -2,51 +2,49 @@ on: [push, pull_request]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       max-parallel: 10
       fail-fast: false
       matrix:
+        os: [ubuntu-latest, ]
```
Comment (sdesrozis) [inline .github/workflows/unittests.yml:10]: What do you think to add `mac-latest` too?

File: .github/workflows/unittests.yml [hunk 2]
```diff
@@ -2,51 +2,49 @@ on: [push, pull_request]
...
 
     - name: Run Tests
+      shell: bash -l {0}
       run: |
-        export PATH="$HOME/miniconda/bin:$PATH"
-        source activate test-environment
         CI_PYTHON_VERSION="${{ matrix.python-version }}" sh tests/run_cpu_tests.sh
 
     - name: Run MNIST Examples
+      if: ${{ matrix.os == 'ubuntu-latest' }}
```
Comment (sdesrozis) [inline .github/workflows/unittests.yml:45]: Why do not run `MNIST` on Windows ? It’s a small test.

File: .github/workflows/unittests.yml (hunk 1 above)
Comment (vfdev-5) [reply .github/workflows/unittests.yml:10]: Maybe in the same as windows...

File: .github/workflows/unittests.yml (hunk 2 above)
Comment (vfdev-5) [reply .github/workflows/unittests.yml:45]: Right, why not

File: tests/ignite/contrib/handlers/test_time_profilers.py [hunk 3]
```diff
@@ -49,6 +52,7 @@ def delay_get_batch_completed(engine):
     return dummy_trainer
 
 
+@pytest.mark.skipif(sys.platform.startswith("mac"), reason="Skip on MacOS")
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_time_profilers.py:55]: `sys.platform` for macosx is "darwin"

Comment (sdesrozis) [reply tests/ignite/contrib/handlers/test_time_profilers.py]: fixed, sorry

**Pull request merged**